### ISSUE TITLE
v2 dat and json toJSON() - adapters give different peaks

### ISSUE THREAD

**Issue by naomiaro** [bbc/waveform-data.js#96]
Hello!

I'm using audiowaveform v1.5.0 to generate both a json and dat file for testing atm.
- same mp3
- same zoom and bits

```
audiowaveform -i Vocals30.mp3 -o test.json -z 1000 -b 16

audiowaveform -i Vocals30.mp3 -o test.dat -z 1000 -b 16
```

I've noticed after using `WaveformData.create()` from both of these loaded sources, and then calling `toJSON()` that the peaks array in the dat file doesn't look correct. The files have matching metadata at the top however, it's just the data array that is quite different.

Are these outputs intended to be identical? I could give this a look.

**Comment by naomiaro** [comment]
The adapter `lib/adapters/arraybuffer.js` only seems to use Int8. When I tried to use 8 bits as a setting, the two arrays still weren't a match however.

`at: function at_sample(index) {
    return this._data.getInt8(this._offset + index);
  },`

So it can't follow the spec here: https://github.com/bbc/audiowaveform/blob/master/doc/DataFormat.md

**Comment by chrisn** [comment]
Yes, embarrassingly the code doesn't handle 16-bit binary data correctly. I'll push a fix + tests shortly. Thanks a lot for reporting this!

**Comment by chrisn** [comment]
I have just published v4.1.1 to npm. Please let me know if this works OK for you.

**Comment by naomiaro** [comment]
Awesome, everything is looking good for me over here now. 👀 

Thank-you!
